compiler/lib/generate.ml

```diff
@@ -285,12 +285,14 @@ module Ctx = struct
     ; should_export : bool
     ; effect_warning : bool ref
     ; cps_calls : Effects.cps_calls
+    ; deadcode_sentinal : Var.t option
```

Comment:
Having this not an option should decrease the number of comparison. The sentinal can be set to a fresh variable if not provided.

Reply:
[6867324](https://github.com/ocsigen/js_of_ocaml/pull/1503/commits/68673240462c4d700ed3d0613bfec51d8e043f95)